### ISSUE TITLE
Missing `security.dll`

### ISSUE THREAD

**Issue by cilerler** [microsoft/Windows-Containers#134]
Can you please add `security.dll` under `C:\Windows\System32`?

So many of us hitting the error message `The instance of SQL Server you attempted to connect to requires encryption, but this machine does not support it.` error upon trying to connect Azure SQL.

e.g.
https://github.com/dotnet/SqlClient/issues/653
https://github.com/dotnet/SqlClient/issues/714

using
```Dockerfile
FROM mcr.microsoft.com/dotnet/runtime:5.0 AS base
```

## Side note
⚠️ _Also, the one in `mcr.microsoft.com/dotnet/sdk:5.0-windowsservercore-ltsc2019` Docker image does not solve the issue, however, using the one in **Windows10** does._

**Comment by vrapolinario** [comment]
Can you please confirm this works with the new Server image? https://techcommunity.microsoft.com/t5/containers/announcing-a-new-windows-server-container-image-preview/ba-p/2304897

**Comment by weijuans-msft** [comment]
@cilerler Also wanted to double check: rre you using .NET Framework not .NET Core right? This thread dotnet/SqlClient#653 shared 2 workarounds for .NET core containers and it seemed working for some folks.

**Comment by cilerler** [comment]
@weijuans-msft, my code is based on .NET Core, and the suggested solution didn't work for me.  Also, as I stated above, even copying `security.dll` in `mcr.microsoft.com/dotnet/sdk:5.0-windowsservercore-ltsc2019` Docker image didn't solve the issue either.  The only solution that did work for me was copying the `security.dll` in `Windows10`.

**Comment by cilerler** [comment]
@vrapolinario Testing base image is not easy in this case.  But I will test it; I just need a little time since that image is a base, and I have to apply what `mcr.microsoft.com/dotnet/runtime` and `mcr.microsoft.com/dotnet/sdk` do on top of it.
I will get back to you.

**Comment by weijuans-msft** [comment]
> @weijuans-msft, my code is based on .NET Core, and the suggested solution didn't work for me. Also, as I stated above, even copying `security.dll` in `mcr.microsoft.com/dotnet/sdk:5.0-windowsservercore-ltsc2019` Docker image didn't solve the issue either. The only solution that did work for me was copying the `security.dll` in `Windows10`.

@richlander do you know this? any reason why?

**Comment by cilerler** [comment]
@weijuans-msft why did you close it?

**Comment by weijuans-msft** [comment]
> @weijuans-msft why did you close it?

There must be an accident. Didn't mean to.

**Comment by jalanubha** [comment]
is there any update on this. Facing same issue with .net core on windows container

**Comment by krokofant** [comment]
~~None of the workarounds listed worked for me. Copying it from `servercore:ltsc2019` or local Windows 10 didn't work. 🤔~~

**Comment by cilerler** [comment]
it should be related to the Windows 10 version, which version you are using

**Comment by krokofant** [comment]
@cilerler Windows 10 20H2 19042.1165
___
Edit: Maybe I'm using the wrong base container? It's still nanoserver base and .net core installed.

```
FROM mcr.microsoft.com/dotnet/aspnet:3.1 as core-runtime
FROM mcr.microsoft.com/powershell:lts-nanoserver-1809

# Install .NET Core
COPY --from=core-runtime ["/Program Files/dotnet", "/Program Files/dotnet"]
USER ContainerAdministrator
RUN setx /M PATH "%PATH%;C:\Program Files\dotnet"
USER ContainerUser
ENV DOTNET_RUNNING_IN_CONTAINER=true
```

**Comment by cilerler** [comment]
DLL should be fine. 
You may review my [DockerFile](https://gist.github.com/cilerler/a1e9768e95dc585eb78376eeb9ed0afb) to figure out what is different on your end.
⚠️  make sure that you have the `security.dll` under the project's `./dist/windows10/System32/` folder.

**Comment by zhiweiv** [comment]
This works for me
```
FROM mcr.microsoft.com/windows/servercore:1809 as core

FROM mcr.microsoft.com/dotnet/sdk:5.0-nanoserver-1809
COPY --from=core /Windows/System32/security.dll /Windows/System32/security.dll
WORKDIR /app
COPY . /app/
CMD ["dotnet", "test", "util.test.e2e.windows.dll", "--nologo"]
```

**Comment by camistein** [comment]
@zhiweiv Your solution worked for me, big thanks!

Tho it is def. not pretty to copy system dll's like that so I really hope this issue can be solved somehow.

**Comment by weijuans-msft** [comment]
Thanks for confirming. We are adding that into our backlog.

**Comment by weijuans-msft** [comment]
For Microsoft Internal tracking, this is the ADO # 37355106.

**Comment by ghost** [comment]
This issue has been open for 30 days with no updates.
@weijuans-msft, @judyliu-ms, please provide an update or close this issue.

**Comment by pkanavos** [comment]
This issue came up again in StackOverflow for the .NET 6 image https://stackoverflow.com/questions/70707378/dockerized-net6-application-throws-microsoft-data-sqlclient-sqlexception-0x801/70710515

```
FROM mcr.microsoft.com/dotnet/aspnet:6.0 AS base
```

**Comment by Saibamen** [comment]
This same issue on `mcr.microsoft.com/dotnet/aspnet:6.0-nanoserver-1809` tag.

But it is **working on** `mcr.microsoft.com/dotnet/aspnet:6.0-windowsservercore-ltsc2019` tag!

**Comment by judyliu-ms** [comment]
Hey folks - we hear you & we are working on getting a solution out the door for you all as soon as possible!

**Comment by gregorybrad** [comment]
Hi @judyliu-ms 
Any ETA on a solution to this, cant get a workaround working on our side...

**Comment by TXP-AlbertPucciani** [comment]
Is there any update on this, we're running into this issue now too

**Comment by KlavsBenislavskis** [comment]
So are there any update on this, we're running into this issue now also

**Comment by krokofant** [comment]
Seems like there is limited effort into improving Windows containers overall.

**Comment by TerrySansom** [comment]
This cannot be the fix, something official would be great

FROM mcr.microsoft.com/windows/servercore:1809 as core

FROM mcr.microsoft.com/dotnet/sdk:6.0-nanoserver-1809
COPY --from=core /Windows/System32/security.dll /Windows/System32/security.dll

**Comment by TerrySansom** [comment]
Hi msfbot - we would like to the msft Dapper team address this issue with an official image.

**Comment by judyliu-ms** [comment]
Hi folks, this should be addressed in the 5.0 release of the SQLclient image.

**Comment by Saibamen** [comment]
@judyliu-ms: Can you link to PR or commit? I can't see it.

Latest 5.0 release of SQLClient is from Oct 11, 2022: https://github.com/dotnet/SqlClient/releases (version 5.0.1: https://github.com/dotnet/SqlClient/releases/tag/v5.0.1)
